src/server/editorServices.ts

```diff
@@ -851,21 +854,31 @@ namespace ts.server {
             return { success: true, projectOptions, configFileErrors: errors };
         }
 
-        private exceededTotalSizeLimitForNonTsFiles<T>(options: CompilerOptions, fileNames: T[], propertyReader: FilePropertyReader<T>) {
+        private exceededTotalSizeLimitForNonTsFiles<T>(name: string, options: CompilerOptions, fileNames: T[], propertyReader: FilePropertyReader<T>) {
             if (options && options.disableSizeLimit || !this.host.getFileSize) {
                 return false;
             }
+
+            let availableSpace = maxProgramSizeForNonTsFiles;
+            this.projectToSizeMap[name] = 0;
+            for (const key in this.projectToSizeMap) {
+                availableSpace -= (this.projectToSizeMap[key] || 0);
+            }
+
             let totalNonTsFileSize = 0;
             for (const f of fileNames) {
                 const fileName = propertyReader.getFileName(f);
                 if (hasTypeScriptFileExtension(fileName)) {
                     continue;
                 }
                 totalNonTsFileSize += this.host.getFileSize(fileName);
-                if (totalNonTsFileSize > maxProgramSizeForNonTsFiles) {
+                if (totalNonTsFileSize > availableSpace) {
+                    this.projectToSizeMap[name] = totalNonTsFileSize;

```

Comment:
Do we disable the language service all up? Or on a project by project basis (seems like the latter from where this function is called). If the latter, then why would be add the size of the files up to where it "broke" the limit, if we're disabling for the whole project? Imagine project A, B, and C in a solution. If B is 25MB, but A & B are 1MB, then after B breaks the limit, it will save that project size here (even though the language service was entirely disabled for B), and then C won't load either.